File: android/src/main/kotlin/com/oversketch/progresshud/ProgresshudPlugin.kt
package com.oversketch.progresshud

import android.util.Log
import com.bigkoo.svprogresshud.SVProgressHUD
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import io.flutter.plugin.common.PluginRegistry.Registrar

lateinit var dialog: SVProgressHUD
lateinit var maskType: SVProgressHUD.SVProgressHUDMaskType

class ProgresshudPlugin: MethodCallHandler {
    companion object {
        @JvmStatic
        fun registerWith(registrar: Registrar) {
            val channel = MethodChannel(registrar.messenger(), "progresshud")
            channel.setMethodCallHandler(ProgresshudPlugin())

            dialog = SVProgressHUD(registrar.activity())
            maskType = SVProgressHUD.SVProgressHUDMaskType.None
        }
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        Log.d("WOW", call.method)
        if (call.method == "getPlatformVersion") {
            result.success("Android ${android.os.Build.VERSION.RELEASE}")
        } else if (call.method == "show") {
            dialog.showWithMaskType(maskType)
        } else if (call.method == "showWithStatus") {
            dialog.showWithStatus(call.arguments.toString(), maskType)
        } else if (call.method == "showInfoWithStatus") {
            dialog.showInfoWithStatus(call.arguments.toString(), maskType)
        } else if (call.method == "showErrorWithStatus") {
            dialog.showErrorWithStatus(call.arguments.toString(), maskType)
        } else if (call.method == "showSuccessWithStatus") {
            dialog.showSuccessWithStatus(call.arguments.toString(), maskType)
        } else if (call.method == "setDefaultMaskTypeNone") {
            maskType = SVProgressHUD.SVProgressHUDMaskType.None
            dialog.showWithMaskType(SVProgressHUD.SVProgressHUDMaskType.None)
        } else if (call.method == "setDefaultMaskTypeBlack") {
            maskType = SVProgressHUD.SVProgressHUDMaskType.Black
            dialog.showWithMaskType(SVProgressHUD.SVProgressHUDMaskType.Black)
        } else if (call.method == "setDefaultMaskTypeClear") {
            maskType = SVProgressHUD.SVProgressHUDMaskType.Clear
            dialog.showWithMaskType(SVProgressHUD.SVProgressHUDMaskType.Clear)
        } else if (call.method == "setDefaultMaskTypeGradient") {
            maskType = SVProgressHUD.SVProgressHUDMaskType.Gradient
            dialog.showWithMaskType(SVProgressHUD.SVProgressHUDMaskType.Gradient)
        } else if (call.method == "dismiss") {
            dialog.dismiss()
        } else if (call.method == "isVisible") {
            if (dialog.isShowing) {
                result.success(true)
            } else {
                result.success(false)
            }
        } else {
            result.notImplemented()
        }
    }
}

File: ios/Classes/ProgresshudPlugin.m
#import "ProgresshudPlugin.h"
#import "SVProgressHUD.h"

@implementation ProgresshudPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"progresshud"
            binaryMessenger:[registrar messenger]];
  [SVProgressHUD setMaximumDismissTimeInterval:2.5];
  [SVProgressHUD setMinimumDismissTimeInterval:1.5];
  ProgresshudPlugin* instance = [[ProgresshudPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getPlatformVersion" isEqualToString:call.method]) {
    result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
  }
    /// show loading
  else if ([@"show" isEqualToString:call.method]) {
      [SVProgressHUD show];
    result(@"success");
  }
//  else if ([@"showProgress" isEqualToString:call.method]) {
//      [SVProgressHUD showProgress:call.arguments];
//      result(@"success");
//  }
    /// show status
  else if ([@"showWithStatus" isEqualToString:call.method]) {
      [SVProgressHUD showWithStatus:call.arguments];
      result(@"success");
  }

  else if ([@"showInfoWithStatus" isEqualToString:call.method]) {
      [SVProgressHUD showInfoWithStatus:call.arguments];
      result(@"success");
  }
  else if ([@"showErrorWithStatus" isEqualToString:call.method]) {
      [SVProgressHUD showErrorWithStatus:call.arguments];
      result(@"success");
  }
  else if ([@"showSuccessWithStatus" isEqualToString:call.method]) {
      [SVProgressHUD showSuccessWithStatus:call.arguments];
      result(@"success");
  }
  /// style
  else if ([@"setDefaultStyleDark" isEqualToString:call.method]) {
      [SVProgressHUD setDefaultStyle:SVProgressHUDStyleDark];
      result(@"success");
  }
  else if ([@"setDefaultStyleLight" isEqualToString:call.method]) {
      [SVProgressHUD setDefaultStyle:SVProgressHUDStyleLight];
      result(@"success");
  }
    /// mask type
  else if ([@"setDefaultMaskTypeNone" isEqualToString:call.method]) {
      [SVProgressHUD setDefaultMaskType:SVProgressHUDMaskTypeNone];
      result(@"success");
  }
  else if ([@"setDefaultMaskTypeBlack" isEqualToString:call.method]) {
      [SVProgressHUD setDefaultMaskType:SVProgressHUDMaskTypeBlack];
      result(@"success");
  }
  else if ([@"setDefaultMaskTypeClear" isEqualToString:call.method]) {
      [SVProgressHUD setDefaultMaskType:SVProgressHUDMaskTypeClear];
      result(@"success");
  }
  else if ([@"setDefaultMaskTypeGradient" isEqualToString:call.method]) {
      [SVProgressHUD setDefaultMaskType:SVProgressHUDMaskTypeGradient];
      result(@"success");
  }
    /// visiable
  else if ([@"isVisible" isEqualToString:call.method]) {
      bool visible = [SVProgressHUD isVisible];
      result(@(visible));
  }
  /// dismiss
  else if ([@"dismiss" isEqualToString:call.method]) {
      [SVProgressHUD dismiss];
      result(@"success");
  }
  /// error
  else {
    result(FlutterMethodNotImplemented);
  }
}

@end
